command/core.py

```diff
@@ -46,3 +50,23 @@ def handle_team_join(self, event_data):
             logging.info(new_id + " added to database - user notified")
         except SlackAPIError:
             logging.error(new_id + " added to database - user not notified")
+
+    def get_help(self):
+        """Get help messages and return a formatted string for messaging."""
+        message = {"text": "Displaying all available commands. To read about"
+                           " a specific command, use `/rocket COMMAND help`\n",
+                   "mrkdwn": "true"}
+        attachments = []
+        for cmd in self.__commands:
+            cmd_name = self.__commands[cmd].get_name()
+            cmd_text = "*[" + cmd_name.upper() + "]*\n\n" + \
+                "Commands for editing", cmd_name + "s."
+            # use below when cmd_usage is fixed in #202
```

Comment:
Hmm, I don't think it'll be a good idea to display the full usage for a command with `/rocket help`, even after #202 is fixed. What might be the best idea here is to add a hardcoded "description" field for each command, then you can concatenate that onto `cmd_name` and just ignore `cmd_usage` (because, as you suggest in the help text intro, users should just run `/rocket COMMAND help` to see usage for a particular command).

Reply:
In particular, that would lead to output like
```
user: interact with users
team: interact with teams
...
```
which I think would be more useful to users than a list of usage for every single subcommand.

Reply:
what is the difference between format_usage and format_help when printing out argparse instructions? i can't really make sense of which is used for what in the examples provided by the docs

Reply:
I'm _guessing_ `format_help` is what we want, though seeing as how I haven't used it before I'm honestly not sure myself 😛 We'll just have to play around with it a bit probably.

That can be done in #202 though, I think.

Reply:
wait so i dont need to change anything else? besides like posting pics of the output

Reply:
Oh, sorry, I think I'm being unclear 😂 Kinda confusing that we have two different levels of help implemented in completely different ways, and I think I also slightly misunderstood your code...

So what I'm getting at (now that I understand it all better 😂) is that you're hardcoding "Commands for editing *blank*". I think what might be a better/more extensible design is to have a `description` field for each command (which might just be "Commands for editing users" or whatever for now - I don't mind if it's hardcoded into the command itself) which is then concatenated on like you have now. We could then later on enforce that all commands actually have the `description` field, but they could have whatever description that makes sense for that command.

So, yep, all the `argparse` stuff isn't necessary to do yet - I would just prefer a `description` field in each command (with getters/setters/whatever) that then gets concatenated on similar to what you have now. You could also remove the note about adding it after #202 - the point here is that `get_help` or whatever we end up having for command-specific help doesn't need to interact with this at all (e.g. if we do `/rocket help`, we don't need to see all the options for `/rocket user`).

Anyway, hopefully that made sense and sorry I'm being confusing!

Reply:
(But yes, please post output from whatever changes you make, if it's convenient - it's a bit hard to tell how the Slack formatting will get translated to actual Slack messages.)

Reply:
okay i think i get it; the description is neither the usage nor the help of each command, just like text to what it does

Reply:
Yep, which is basically what you already have with the "Commands for editing users" line... I just want each command to have its own description 😛